Documentation/Evolution/StronglyTypedCaptures.md

```diff
@@ -596,9 +596,73 @@ included in the generic signature.
 Given all of this, it seems simpler and more pragmatic to make `Regex` generic
 over both the match and the captures.
 
-### Nested rather than flattened captures
+### Structured rather than flat captures
 
-TODO: Including a more ergonomic Alternation. Maybe the default for `Pattern`?
+This pitch proposes infering capture types in such a way as to align with the
+traditional numbering of backreferences. This is because much of the motivation
+behind providing regex literals in Swift is their familiarity.
+
+If we decided to deprioritize this motivation, there are opportunities to infer
+safer, more ergonomic, and arguably more intuitive types for captures.
+
+For example, to be consistent with traditional regex backreferences
+quantifications of multiple or nested captures produces parallel arrays rather
+than an array of tuples.
+
+```swift
+/(?:(?<lower>[0-9a-fA-F]+)\.\.(?<upper>[0-9a-fA-F]+))+/
+// Flat captures type:
+// => `Regex<(Substring, lower: [Substring], upper: [Substring])>`
+
+// Structured captures type:
+// => `Regex<(Substring, [(lower: Substring, upper: Substring)])>`
+```
+
+The structured captures type is safer because the type system encodes that there
```

Comment:
There are occurrences of both "captures type" and "capture type". Let's standardize on "capture type"?